libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/couchdb/DatabaseRepository.java

```diff
@@ -97,11 +97,28 @@ public DatabaseRepository(Class<T> type, DatabaseConnector databaseConnector) {
     }
 
     public Set<String> queryForIdsAsComplexValues(String queryName, Map<String, Set<String>> keys) {
-        Set<ComplexKey> complexKeys = keys.entrySet().stream()
-                .map(DatabaseRepository::createComplexKeys)
-                .flatMap(Collection::stream).collect(Collectors.toSet());
-        ViewQuery query = createQuery(queryName).keys(complexKeys);
-        return queryForIds(query);
+        Set<String> queryResult = new HashSet<>();
+        Set<ComplexKey> complexKeys = new HashSet<>();
+
+        for (Map.Entry<String, Set<String>> externalId : keys.entrySet()) {
+            Set<String> externalValues = externalId.getValue();
+            if (externalValues.isEmpty() || (externalValues.size() == 1 && externalValues.iterator().next().isEmpty())) {
```

Comment:
But now one can no longer search for all entries with key type `someExternalIdName` and empty value?

I think this implicit behaviour would better be moved to another explicit function.

Reply:
I agree with the first bit about searching for empty values, this is valuable and should not be _lost_.

However I would suggest an alternative solution approach:
- _**.../searchByExternalIds?external-id-key**_ - search only by key
- _**.../searchByExternalIds?external-id-key=**_ - search by key and value (in this case, an empty value)